util/sqlutil/splitquery.go

```diff
@@ -31,15 +32,19 @@ func sqlSplitBlock(blockIdx int, data []byte, atEOF bool) (advance int, token []
 	return next + advance, data[:next+len(token)], nil
 }
 
+// https://www.postgresql.org/docs/16/sql-syntax-lexical.html#SQL-SYNTAX-DOLLAR-QUOTING
+var splitRx = regexp.MustCompile(`\$[^$]*\$`)
+
 func sqlSplitQuery(data []byte, atEOF bool) (advance int, token []byte, err error) {
 	if atEOF && len(data) == 0 {
 		return 0, nil, nil
 	}
 	semiIdx := bytes.IndexRune(data, ';')
-	blockIdx := bytes.Index(data, []byte("$$"))
-	if blockIdx != -1 && (semiIdx == -1 || semiIdx > blockIdx) {
+
+	idx := splitRx.FindIndex(data)
+	if idx != nil && (semiIdx == -1 || semiIdx > idx[0]) {
 		// have block start and it comes before semi (or no semi)
-		return sqlSplitBlock(blockIdx, data, atEOF)
+		return sqlSplitBlock(data[idx[0]:idx[1]], idx[0], data, atEOF)
```

Comment:
we should probably have variable names that are more self-explanatory. Or comments that explain why we are doing this slice operation in place.